crates/cairo-addons/src/vm/dict_manager.rs

```diff
@@ -75,4 +149,33 @@ impl PyDictTracker {
             Ok(Self { inner: DictTracker::new_with_initial(current_ptr.inner, data) })
         }
     }
+
+    #[getter]
+    fn current_ptr(&self) -> PyRelocatable {
+        PyRelocatable { inner: self.inner.current_ptr }
+    }
+
+    #[getter]
+    fn data(&self) -> HashMap<PyDictKey, PyMaybeRelocatable> {
+        self.inner
+            .get_dictionary_ref()
+            .iter()
+            .map(|(k, v)| (PyDictKey::from(k.clone()), PyMaybeRelocatable::from(v.clone())))
```

Comment:
why are we cloning entire dictionaries when getting them?

Reply:
the values are behind shared references and need to be moved to the python objects.